src/service-worker.ts

```diff
@@ -59,12 +59,14 @@ function getBundledPrecachedAssets(project: PolymerProject) {
  * contents), based off of the options provided.
  */
 export function generateServiceWorker(options: AddServiceWorkerOptions): Promise<Buffer> {
+  console.assert(!!options, '`project` & `buildRoot` options are required');
   console.assert(!!options.project, '`project` option is required');
   console.assert(!!options.buildRoot, '`buildRoot` option is required');
 
+  options = Object.assign({}, options);
   let project = options.project;
   let buildRoot = options.buildRoot;
-  let swConfig: SWConfig = options.swConfig || {};
+  let swConfig: SWConfig = Object.assign({}, options.swConfig);
 
   return project.analyzer.analyzeDependencies.then((depsIndex: DepsIndex) => {
     let staticFileGlobs = swConfig.staticFileGlobs || [];
```

Comment:
we actually never modify `swConfig.staticFileGlobs`. Instead we return new arrays via concat and then overwrite the `staticFileGlobs` property with one of those new arrays. So this isn't an issue today.

BUT you're right, code changes and this might not be the case forever. I'll add this extra precaution back in.